src/app/modules/main/chat_section/chat_content/messages/module.nim

```diff
@@ -232,9 +236,37 @@ method reevaluateViewLoadingState*(self: Module) =
                        self.firstUnseenMessageState.fetching or
                        self.view.getMessageSearchOngoing())
 
+proc generateAlbumIdToImageMap(self: Module, messages: seq[MessageDto]): Table[string, seq[string]] =
+  var albumIdToImagesMap = initTable[string, seq[string]]()
+
+  for message in messages:
+    if message.albumId in albumIdToImagesMap:
+      albumIdToImagesMap[message.albumId].add(message.image)
+    else:
+      albumIdToImagesMap[message.albumId] = @[message.image]
+  
+  return albumIdToImagesMap
+
+
+proc setQuotedMessageImages(self: Module, messages: seq[MessageDto], items: var seq[Item], albumIdToImagesMap: Table[string, seq[string]]) =
+  for message in messages:
+    for i in 0 ..< items.len:
+      let item = items[i]
+      
+      var quotedMessageAlbumMessageImages = item.quotedMessageAlbumMessageImages
+
+      if message.id == item.responseToMessageWithId:
+        if message.albumId in albumIdToImagesMap:
+          quotedMessageAlbumMessageImages = albumIdToImagesMap[message.albumId]
+          item.quotedMessageAlbumMessageImages = quotedMessageAlbumMessageImages
+          item.quotedMessageAlbumImagesCount = quotedMessageAlbumMessageImages.len
+          items[i] = item
+
 method newMessagesLoaded*(self: Module, messages: seq[MessageDto], reactions: seq[ReactionDto]) =
   var viewItems: seq[Item]
 
+  var albumIdToImagesMap = self.generateAlbumIdToImageMap(messages)
```

Comment:
I think using only the messages received here might not generate a good map, because `newMessagesLoaded` only contains ~20 messages. In most cases, it should be ok, but if someone replied to an old message, then the album of messages wouldn't be part of the messages `seq`, so we'd also not find the images.

I think for now, this solution is still the right one as it's simple and does the trick for most of the cases, but we should open a new issue to actually get the images in status-go directly when we get the messages. That way, we have access to the whole DB and not just a sample of the messages.

Maybe we can add a TODO here with a link to the issue to remember.

Reply:
That's a very good point I thought about too. And I agree that for now the mitigation is that it's probably not happening in most cases (maybe data should back us here) and for sure I need to add a TODO with another issue as of now.